src/modules/user-auth/utils/AuthGuard/AuthGuard.tsx

```diff
@@ -0,0 +1,38 @@
+import { FC, ReactNode, useEffect } from 'react';
+import { useAuth } from '@modules/user-auth/contexts/AuthContext';
+import DefaultUnauthenticatedPrompt from '../../components/DefaultUnauthenticatedPrompt';
+import DefaultEmailUnverifiedPrompt from '../../components/DefaultEmailUnverifiedPrompt';
+import { useLayoutModal } from '@modules/layouts/contexts/LayoutModalContext';
+
+type AuthGuardProps = {
+  emailVerified?: boolean;
+  promptLogin?: boolean;
+  renderUnauthenticated?: ReactNode;
+  renderEmailUnverified?: ReactNode;
+};
+
+const AuthGuard: FC<AuthGuardProps> = ({
+  children,
+  emailVerified = false,
+  promptLogin = false,
+  renderUnauthenticated = <DefaultUnauthenticatedPrompt />,
```

Comment:
May want to change this to `renderedUnauthenticated` and `renderEmailUnverified` because `render` hints a function.